File: manpower-blog-frontend/src/types/system/user/userResponse.ts
import type { AccountStatus, AccountType, VerifiedStatus } from "@/types/enums/account"
import type { Status } from "@/types/enums/status"

export interface UserVO {
    /**
     * ユーザーID
     */
    userId: number

    accountId: number


    /**
     * ニックネーム
     */
    nickName: string

    /**
     * ユーザー状態
     */
    userStatus: Status

    /**
     * アカウントタイプ
     */
    accountType: AccountType

    /**
     * アカウント値
     */
    accountValue: string

    /**
     * アカウント状態
     */
    accountStatus: AccountStatus

    /**
     * 認証状態
     */
    verifiedStatus: VerifiedStatus

    /**
     * 作成日時
     */
    createdAt: string

    roleId: number

    roleName: string

}

export interface UserView extends UserVO {
  /**
   * ローディング状態
   */
  _loading: boolean   
}

File: manpower-blog-backend/blog-module-system/src/main/java/com/manpowergroup/blog/module/system/application/dto/response/user/UserResponse.java
package com.manpowergroup.blog.module.system.application.dto.response.user;

import com.manpowergroup.blog.shared.enums.Status;
import com.manpowergroup.blog.shared.enums.VerifiedStatus;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

@Schema(description = "ユーザー情報")
public record UserResponse(
        @Schema(description = "ユーザーID") Long userId,
        @Schema(description = "アカウントID") Long accountId,
        @Schema(description = "ニックネーム") String nickName,
        @Schema(description = "ユーザーステータス") Status userStatus,
        @Schema(description = "アカウント種別") String accountType,
        @Schema(description = "アカウントログインID") String accountValue,
        @Schema(description = "アカウントステータス") Status accountStatus,
        @Schema(description = "アカウント認証状態") VerifiedStatus verifiedStatus,
        @Schema(description = "ユーザー作成日時") LocalDateTime createdAt,
        @Schema(description = "ロールID") Long roleId,
        @Schema(description = "ロール名") String roleName
) {
}
